File: 5Days-SoftwareArchitecture/typesript/Day2/Connascence/con_name.ts
class BankService {
    deposit(amount: number) {
        console.log(`Deposited ₹${amount}`);
    }
}

class Customer {
    private bank = new BankService();

    doDeposit() {
        this.bank.deposit(1000);
    }
}

new Customer().doDeposit();

File: 5Days-SoftwareArchitecture/Java/Day2/Connascence/CoN_Name.java
class BankService {
    void deposit(double amount) {        // method name: deposit
        System.out.println("Deposited ₹" + amount);
    }
}

class Customer {
    private BankService bank = new BankService();

    void doDeposit() {
        bank.deposit(1000);              // must use same name 'deposit'
    }
}

public class CoN_Name {
    public static void main(String[] args) {
        new Customer().doDeposit();
    }
}
